src/tests/JIT/Regression/JitBlue/Runtime_86027/Runtime_86207.cs

```diff
@@ -0,0 +1,87 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+using Xunit;
+
+public class Runtime_86027
+{
+    [Fact]
+    public static int TestRotateLeft1()
+    {
+        long[] firstOp = new long[] { 7822136809956075968, 1 };
```

Comment:
presumably nothing prevents us from folding `firstOp[0]` to a constant and then perform constant folding in VN so the actual instruction will never be emitted

Reply:
Right, but the bug repros in T0 and Debug, so it should get hit in general unless we start doing those foldings there as well

Reply:
it sounds that we waste TP in Tier0 to recognize this pattern

Reply:
The bug repro'd in T0 because we recognize `long.RotateRight` in import. It's a primitive intrinsic which is trivial to recognize, much like the other ones we recognize in T0.

Reply:
> it's a primitive intrinsic which is trivial to recognize

Ah ok, I though it's about that bigger one in morph that recognizes plain shifts